### ISSUE TITLE
exception when using an boolean index array that is all false

### ISSUE THREAD

**Issue by davidslac** [h5py/h5py#717]
If you run this code

```
import numpy as np
import h5py
h5 = h5py.File('test.h5','w')
h5['arr'] = np.zeros((5,3,10))
h5.close()
idx = np.zeros(5).astype(np.bool)
h5py.File('test.h5','r')['arr'][idx,0,:]

```

with h5py version 2.5.0 you get

```
Traceback (most recent call last):
  File "empty_index.py", line 7, in <module>
    h5py.File('test.h5','r')['arr'][idx,0,:]
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper (/reg/g/psdm/sw/dist/rpms/BUILD/h5py-2.5.0b/h5py-2.5.0/h5py/_objects.c:2880)
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper (/reg/g/psdm/sw/dist/rpms/BUILD/h5py-2.5.0b/h5py-2.5.0/h5py/_objects.c:2839)
  File "/reg/g/psdm/sw/releases/ana-current/arch/x86_64-rhel7-gcc48-opt/python/h5py/_hl/dataset.py", line 431, in __getitem__
    selection = sel.select(self.shape, args, dsid=self.id)
  File "/reg/g/psdm/sw/releases/ana-current/arch/x86_64-rhel7-gcc48-opt/python/h5py/_hl/selections.py", line 95, in select
    sel[args]
  File "/reg/g/psdm/sw/releases/ana-current/arch/x86_64-rhel7-gcc48-opt/python/h5py/_hl/selections.py", line 462, in __getitem__
    mshape = list(count)
UnboundLocalError: local variable 'count' referenced before assignment
```

That is, if you use an index array that is all false, you get an unboundLocalError for a variable called count. Maybe return an empty array instead?

**Comment by swyoon** [comment]
I have encountered the same error and would like to contribute fixing it.
Will it be too difficult?

**Comment by chr5tphr** [comment]
This is the same as #1169 which was fixed in #1174. Thus, this issue can be closed.

**Comment by dhamelse** [comment]
> The underlying problem was the same as in #1169 which was fixed in #1174. Thus, this issue can be closed.

Hello, I'm using h5py 2.9.0 on pytohn 2.7 for research. We must use python 2.7 because other packages we are incorporating only support python 2.7. This issue persists for me. Is there a workaround I can try? I know the issue was fixed in later releases but I'm wondering if h5py 2.9 was not fixed. Thank you.
